src/xcode/ENA/ENA/Source/Services/Exposure Transaction/ExposureDetectionExecutor.swift

```diff
@@ -168,11 +168,26 @@ final class ExposureDetectionExecutor: ExposureDetectionDelegate {
 			writtenPackages: WrittenPackages,
 			completion: @escaping (Result<ENExposureDetectionSummary, Error>) -> Void
 	) -> Progress {
+
+		// Clear the key packages and app config on ENError = 2 = .badParameter
+		// For more details, see: https://jira.itc.sap.com/browse/EXPOSUREAPP-3297
+		func clearCacheOnError2(error: Error) {
+			if let enError = error as? ENError, enError.code == .badParameter {
+				// Clear the key packages
+				downloadedPackagesStore.reset()
+				downloadedPackagesStore.open()
+
+				// Clear the app config
+				store.appConfig = nil
```

Comment:
Just to make sure, every change to the store gets instantly persisted to Disk?

Reply:
Its reseting the app config cache. Also added etag and date reset.

Reply:
Its persisted here: https://github.com/corona-warn-app/cwa-app-ios/blob/3c8416d309651783b2c1dafff446661bcac14e9d/src/xcode/ENA/ENA/Source/Workers/SQLiteKeyValueStore.swift#L138